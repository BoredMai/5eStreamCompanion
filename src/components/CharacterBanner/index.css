:local(.character-banner) {
  border-image: url('/images/border_27slice.png') 27;
  border-style: solid;
  border-width: 27px;
  font-size: 2.4rem;
  padding: 1.5rem;
  width: fit-content;
}

:local(.character-banner .bottom-row) {
  display: flex;
  margin-top: 3rem;
}

:local(.character-banner .top-row) {
  display: flex;
}

:local(.character-banner .left-column) {
  display: flex;
  flex-grow: 3;
  flex-direction: column;
}

:local(.character-banner .right-column) {
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  margin-left: 1rem;
}

:local(.character-banner .name) {
  font-size: 3.2rem;
}

:local(.character-banner .attribute-container) {
  display: flex;
}

:local(.character-banner .attribute-container > *:not(:last-child)) {
  margin-right: 1.5rem;
}
