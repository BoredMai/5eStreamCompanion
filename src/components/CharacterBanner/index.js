import cn from 'classnames';
import React, { Component, Fragment } from 'react';

import ACShield from '../ACShield';
import AttributeBlock from '../AttributeBlock';
import FillableBar from '../FillableBar';

import {
  acShieldContainer,
  attributeContainer,
  bottomRow,
  characterBanner,
  charClass,
  flexColumn,
  marginTopHalf,
  name,
  playerName,
  race,
  topRow,
} from './index.css';

export default class CharacterBanner extends Component {
  renderHeader() {
    return (
      <Fragment>
        <div className={topRow}>
          <div className={flexColumn}>
            <span className={name}>Character Name</span>
            <span className={playerName}>Player Name</span>
          </div>
          <div className={flexColumn}>
            <FillableBar color={'#F66'} current={30} label={'HP'} max={60} />
            <FillableBar
              className={marginTopHalf}
              color={'#66F'}
              current={30}
              label={'XP'}
              max={60}
            />
          </div>
          <div className={flexColumn}>
            <span className={race}>Character Race</span>
            <span className={charClass}>Character Class</span>
          </div>
        </div>
      </Fragment>
    );
  }

  render() {
    return (
      <div className={characterBanner}>
        {this.renderHeader()}
        <div className={bottomRow}>
          <div className={attributeContainer}>
            <AttributeBlock attribute={'STR'} value={20} />
            <AttributeBlock attribute={'DEX'} value={20} />
            <AttributeBlock attribute={'CON'} value={20} />
            <AttributeBlock attribute={'INT'} value={20} />
            <AttributeBlock attribute={'WIS'} value={20} />
            <AttributeBlock attribute={'CHA'} value={20} />
          </div>
          <div className={cn(acShieldContainer)}>
            <ACShield value={20} />
          </div>
        </div>
      </div>
    );
  }
}
