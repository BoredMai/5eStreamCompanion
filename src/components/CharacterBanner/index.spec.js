import React from 'react';
import { shallow } from 'enzyme';
import {
  JustName,
  NoShortName,
  PlayerOne,
} from '../../../__tests__/mocks/charMock';
import CharacterBanner from '.';

describe('<CharacterBanner />', () => {
  let wrapper;

  it('renders properly', () => {
    wrapper = shallow(<CharacterBanner {...PlayerOne} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  describe('Battle Mode', () => {
    it('renders properly on battle mode', () => {
      wrapper = shallow(<CharacterBanner {...PlayerOne} battleMode />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('displays short name when defined', () => {
      wrapper = shallow(<CharacterBanner {...PlayerOne} battleMode />);
      const nameTag = wrapper.find('span[className="nameTag"]');
      expect(nameTag.text()).toBe('One');
    });

    it("displays first name when short name isn't defined", () => {
      wrapper = shallow(<CharacterBanner {...NoShortName} battleMode />);
      const nameTag = wrapper.find('span[className="nameTag"]');
      expect(nameTag.text()).toBe('Name');
    });

    it('displays name properly on single word name', () => {
      wrapper = shallow(<CharacterBanner {...JustName} battleMode />);
      const nameTag = wrapper.find('span[className="nameTag"]');
      expect(nameTag.text()).toBe('Monika');
    });
  });
});
