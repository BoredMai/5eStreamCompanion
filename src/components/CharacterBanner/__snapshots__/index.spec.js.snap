// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterBanner /> Battle Mode renders properly on battle mode 1`] = `
<div
  className="borderContainer"
>
  <div
    className="characterBanner"
  >
    <div
      className="flexColumn"
    >
      <span
        className="nameTag"
      >
        One
      </span>
      <FillableBar
        color="#F66"
        current={10}
        label="HP"
        max={10}
      />
    </div>
  </div>
</div>
`;

exports[`<CharacterBanner /> renders properly 1`] = `
<div
  className="borderContainer"
>
  <div
    className="characterBanner"
  >
    <React.Fragment>
      <div
        className="topRow"
      >
        <div
          className="flexBasisHalf flexColumn"
        >
          <span
            className="nameTag"
          >
            Character One
          </span>
          <span
            className="playerTag"
          >
            Player One
          </span>
        </div>
        <div
          className="flexColumn"
        >
          <FillableBar
            color="#F66"
            current={10}
            label="HP"
            max={10}
          />
          <FillableBar
            className="marginTopHalf"
            color="#66F"
            current={0}
            label="XP"
            max={300}
          />
        </div>
        <div
          className="flexColumn"
        >
          <span
            className="raceTag"
          >
            Race
          </span>
          <span
            className="classTag"
          >
            Class 1
          </span>
        </div>
      </div>
    </React.Fragment>
    <div
      className="bottomRow"
    >
      <div
        className="attributeContainer"
      >
        <AttributeBlock
          attribute="STR"
          value={10}
        />
        <AttributeBlock
          attribute="DEX"
          value={10}
        />
        <AttributeBlock
          attribute="CON"
          value={10}
        />
        <AttributeBlock
          attribute="INT"
          value={10}
        />
        <AttributeBlock
          attribute="WIS"
          value={10}
        />
        <AttributeBlock
          attribute="CHA"
          value={10}
        />
      </div>
      <div
        className="acShieldContainer"
      >
        <ACShield
          value={16}
        />
      </div>
    </div>
  </div>
</div>
`;
